AmplicaLabs/social-app-template
# Replace services/graph.ts with graph-service

# Purpose
The goal of this PR is integrate the graph service with the social app template.

Closes #35 

## Solution

- [x] Replace any polkadot.js calls with Graph-Service endpoints.

Started by/Contributions from:
@claireolmstead @JoeCap08055 @saraswatpuneet 

### Change summary
- [x] Added configuration items needed for `graph-service`
- [x] Refactored `GraphService` class
- [x] Updated `docker-compose.yaml` to correctly launch `graph-service-api` and `graph-service-worker`
- [x] Migrated all `graph/dsnpId` endpoints to `graph/msaId`
- [x] Implement scaffolding for graph-service webhook 
      (need to address in follow-on issue #51 )

## Steps to Verify
1. Setup the containers with `docker compose up -d`
2. Sign up/sign in with SIWF
3. Create a post
4. Logout with current user
5. Sign up/sign in with new user account.
6. Navigate to the original user's post.
7. Verify that the follow button is available and executes correctly (there is a delay after the follow request)

## Follow Up

- [ ] #51
- [ ] #52
- [ ] #53
